### PR TITLE
Avoid mapping text in diagnostics from XML doc comment compiler

### LINKED ISSUE #8807: Failing Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests
Affected PRs
- https://github.com/dotnet/roslyn/pull/8802
- https://github.com/dotnet/roslyn/pull/8796
- https://github.com/dotnet/roslyn/pull/8812
- https://github.com/dotnet/roslyn/pull/8801
- https://github.com/dotnet/roslyn/pull/8799
- https://github.com/dotnet/roslyn/pull/8646
- https://github.com/dotnet/roslyn/pull/8435

Test Result (12 failures / +12)
- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationModeTests.XmlSyntaxError_Included
- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationCommentCompilerTests.IncludedName_DuplicateNameAttribute
- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationCommentCompilerTests.PragmaDisableWarningInXmlFile

- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationCommentCompilerTests.WRN_XMLParseIncludeError_Source
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XDocument_WRN_XMLDocInvalidXMLFragment
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XmlError_WRN_XMLDocBadFormedXML
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Bug684184
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_IOError_WRN_XMLDocBadFormedXML
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_FileNotFound_WRN_XMLDocBadFormedXML
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XPathNotFound_WRN_XMLDocInvalidXMLFragment
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_Cycle_WRN_XMLDocInvalidXMLFragment
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XPathError_WRN_XMLDocBadFormedXML

### PR DESCRIPTION
**Pull request by cston**
Fixes #8807

### CODE REVIEW

**Comment by cston** [comment]
@dotnet/roslyn-compiler, @amcasey please review

File: src/Compilers/VisualBasic/Test/Symbol/DocumentationComments/DocCommentTests.vb [hunk 1]
```diff
@@ -12057,7 +12057,7 @@ xmlDoc)
             Return list.ToArray()
         End Function
 
-        Friend Shared Function CompileCheckDiagnosticsAndXmlDocument(
+        Private Shared Function CompileCheckDiagnosticsAndXmlDocument(
```
Comment (amcasey) [inline src/Compilers/VisualBasic/Test/Symbol/DocumentationComments/DocCommentTests.vb]: Is `stringMapper` still used?

**Comment by amcasey** [comment]
LGTM

File: src/Compilers/VisualBasic/Test/Symbol/DocumentationComments/DocCommentTests.vb (hunk 1 above)
Comment (cston) [inline src/Compilers/VisualBasic/Test/Symbol/DocumentationComments/DocCommentTests.vb]: Yes, `stringMapper` is used in the call to `CheckXmlDocument`.

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**